### PR TITLE
Implement Template search

### LINKED ISSUE #426: Implement template search
**Is your feature request related to a problem? Please describe.**
The scope of the issue is to implement a search of templates by name [here](https://github.com/cyclops-ui/cyclops/blob/main/cyclops-ui/src/components/pages/TemplateStore/TemplateStore.tsx).

**Describe the solution you'd like**
You can check out the pod search in [node details](https://github.com/cyclops-ui/cyclops/blob/main/cyclops-ui/src/components/pages/NodeDetails.tsx) and implement it similarly.

### PR DESCRIPTION
**Pull request by kom-senapati**
closes #426 

## 📑 Description
Implemented Template Search

## ✅ Checks
- [ ] I have updated the documentation as required
- [X] I have performed a self-review of my code

## ℹ Additional context
Took reference from modules (implemented similar search bar)

### CODE REVIEW

File: cyclops-ctrl/go.mod [hunk 1]
```diff
@@ -30,7 +30,6 @@ require (
 require (
```
Comment (petar-cvit) [inline cyclops-ctrl/go.mod:30]: @kom-senapati can you try deleting the go.mod file from this line down, and then run `go mod tidy`?

Comment (kom-senapati) [reply cyclops-ctrl/go.mod:30]: Some versions were updated

Comment (petar-cvit) [reply cyclops-ctrl/go.mod:30]: You can push it to this branch. We don't want changes in the go.mod since nothing changes in the go code

Comment (kom-senapati) [reply cyclops-ctrl/go.mod]: Reverted these 2 files to latest commits

File: cyclops-ui/src/components/pages/TemplateStore/TemplateStore.tsx [hunk 2]
```diff
@@ -182,8 +193,18 @@ const TemplateStore = () => {
...
+            placeholder={"Search templates"}
+            style={{ width: "30%" }}
+            onChange={handleSearch}
+          ></Input>
+        </Col>
+      </Row>
+      <Divider orientationMargin="0" />
       <Col span={24} style={{ overflowX: "auto" }}>
-        <Table dataSource={templates}>
+        <Table dataSource={filteredTemplates}>
```
Comment (petar-cvit) [inline cyclops-ui/src/components/pages/TemplateStore/TemplateStore.tsx:207]: Can we remove the divider and add some padding to the searchbar?
```html
      <Row gutter={[40, 0]} style={{paddingBottom: "16px"}}>
        <Col span={18}>
          <Input
            placeholder={"Search templates"}
            style={{ width: "30%" }}
            onChange={handleSearch}
          ></Input>
        </Col>
      </Row>
```

**Review by petar-cvit — COMMENTED** [review]
Thanks @kom-senapati, left one nit pick 😄

**Comment by petar-cvit** [comment]
Also, @kom-senapati, there is a conflict because a new PR was recently merged

File: cyclops-ui/src/components/pages/TemplateStore/TemplateStore.tsx (hunk 2 above)
Comment (kom-senapati) [reply cyclops-ui/src/components/pages/TemplateStore/TemplateStore.tsx:207]: Ya we can do that, check if this much spacing ok

image

Comment (petar-cvit) [reply cyclops-ui/src/components/pages/TemplateStore/TemplateStore.tsx:207]: @kom-senapati yup, looks good

**Review by petar-cvit — APPROVED** [review]
Thanks @kom-senapati 🧡

**Pull request merged**